src/util/checkInput.ts

```diff
@@ -171,15 +172,13 @@ export function checkInput(
     }
   }
 
-  optimization.parameters = parameters;
```

Comment:
@lpatiny this function is a mess. It was modifying the options object, and I don't understand the code above to the point I just put `any` in the types.